interfaces/builtin/cups_control_test.go

```diff
@@ -0,0 +1,188 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type cupsControlSuite struct {
+	iface            interfaces.Interface
+	coreSlotInfo     *snap.SlotInfo
+	coreSlot         *interfaces.ConnectedSlot
+	plugInfo         *snap.PlugInfo
+	plug             *interfaces.ConnectedPlug
+	providerSlotInfo *snap.SlotInfo
+	providerSlot     *interfaces.ConnectedSlot
+}
+
+var _ = Suite(&cupsControlSuite{iface: builtin.MustInterface("cups-control")})
+
+const cupsControlConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [cups-control]
+`
+
+const cupsControlCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  cups-control:
+`
+
+const cupsControlProviderYaml = `name: provider
+version: 0
+apps:
+ app:
+  slots: [cups-control]
+`
+
+func (s *cupsControlSuite) SetUpTest(c *C) {
+	s.plug, s.plugInfo = MockConnectedPlug(c, cupsControlConsumerYaml, nil, "cups-control")
+	s.coreSlot, s.coreSlotInfo = MockConnectedSlot(c, cupsControlCoreYaml, nil, "cups-control")
+	s.providerSlot, s.providerSlotInfo = MockConnectedSlot(c, cupsControlProviderYaml, nil, "cups-control")
+}
+
+func (s *cupsControlSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
```

Comment:
same comment as the other PR, this could be done with `testutil.BaseTest` and `AddCleanup` but this is a minor nitpick

Reply:
Actually, TearDown isn't needed at all since we aren't calling dirs.SetRootDir on a temp dir or anything, so I've removed this.